src/ApiService/Cart/addCart.js

```diff
@@ -5,42 +5,78 @@ export const addCart = async (
   product,
   token,
   CartDispatch,
-  ProductDispatch
+  ProductDispatch,
+  CartState
 ) => {
-  try {
-    const {
-      data: { cart },
-      status,
-    } = await axios.post(
-      "/api/user/cart",
-      { product },
-      {
-        headers: {
-          authorization: token,
-        },
-      }
-    );
-    if (status === 200 || status === 201) {
-      CartDispatch({ type: "ADD_TO_CART", payload: cart });
+  const findItem = CartState.cart.find((item) => item._id === product._id);
 
-      ProductDispatch({
-        type: "DECREASE_PRODUCT_QTY",
-        payload: product._id,
-      });
+  if (findItem === undefined) {
+    try {
+      const {
+        data: { cart },
+        status,
+      } = await axios.post(
```

Comment:
I like this destructuring part. I was using response.data.cart.

Reply:
thanks